lib/strings.nix

```diff
@@ -438,13 +438,53 @@ rec {
        => true
        isStorePath pkgs.python
        => true
+       isStorePath ./types.nix
+       => true
        isStorePath [] || isStorePath 42 || isStorePath {} || …
        => false
   */
   isStorePath = x:
-       builtins.isString x
-    && builtins.substring 0 1 (toString x) == "/"
-    && dirOf (builtins.toPath x) == builtins.storeDir;
+    let
+      toStringContext = y: "${y}";
+    in (builtins.isString x || isPath x)
+    && builtins.substring 0 1 (toStringContext x) == "/"
+    && dirOf (toStringContext x) == builtins.storeDir;
+
+  /* Check whether a string or a path is in the store.
```

Comment:
Same as above, it’s really a check for any value. Otherwise you have to omit the `isString`/`isPath` check.